src/core/Akka.Persistence/Eventsourced.Recovery.cs

```diff
@@ -124,89 +154,105 @@ private EventsourcedState RecoveryStarted(long maxReplays)
         private EventsourcedState Recovering(Receive recoveryBehavior, TimeSpan timeout)
         {
             // protect against event replay stalling forever because of journal overloaded and such
-            var timeoutCancelable = Context.System.Scheduler.ScheduleTellRepeatedlyCancelable(timeout, timeout, Self,
-                new RecoveryTick(false), Self);
+            var timeoutCancelable = Context.System.Scheduler.ScheduleTellRepeatedlyCancelable(timeout, timeout, Self, new RecoveryTick(false), Self);
             var eventSeenInInterval = false;
 
             return new EventsourcedState("replay started", true, (receive, message) =>
             {
-                if (message is ReplayedMessage)
+                try
                 {
-                    var m = (ReplayedMessage)message;
-                    try
+                    if (message is ReplayedMessage)
```

Comment:
I refrained from refactoring anything in these classes not related to the original PR.